Zina_Yeghiazaryan/Web/JS/17_06_2021/pagination.js

```diff
@@ -0,0 +1,118 @@
+class Pagination {
+    _items = [];
+    _pageSize = 10;
+    _count = 100;
+	
+    constructor(items, pageSize, count) {
+        this._items = items;
+        this._pageSize = pageSize;
+        this._count = count;
+    }
+	
+    goToPage(pageNumber) {
+        let index = 0;
+        let array = [];
+		
+        for (let i = (pageNumber-1)*this._pageSize; i < pageNumber*this._pageSize; ++i) {
+			      if (this._items[i] === undefined) {
+				        break;
+			      }
+            array[index] = this._items[i];
+            ++index;
+        }
+		
+        return array;
+    }
+	
+	  prevPage = (pageNumber) => this.goToPage(pageNumber - 1);
+	  nextPage = (pageNumber) => this.goToPage(pageNumber + 1);
+	  firstPage = () => this.goToPage(1);
+	  lastPage = () => this.goToPage(_pageSize);
+	
+	  onClick() {
+		    let k;
+
+		    if (this._count % this._pageSize === 0) {
+		  	    k = this._count / this._pageSize;
+		    } else {
+			      k = this._count / this._pageSize + 1;
+		    }
+		
+		    var num;
+		
+		    var prev = document.createElement("button");
+		    prev.onclick = function() {
+				    document.querySelectorAll("p").forEach(e => e.parentNode.removeChild(e));
+				    var p = document.createElement("p");
+				    p.innerHTML = pagination.prevPage(num);
+				    --num;
+				    document.body.appendChild(p);
+			  };
+			  prev.type = "button";
+			  prev.innerHTML = "previous";
+			  document.body.appendChild(prev);
+		
+		    for (let i = 1; i <= k; ++i) {
+			      if (i == 1) {
```

Comment:
This part contains a lot of duplicate code. We can organize this using a function.